src/viur/core/i18n.py

```diff
@@ -227,38 +235,56 @@ def __str__(self) -> str:
         if self.translationCache is None:
             global systemTranslations
 
-            if self.key not in systemTranslations and conf.i18n.add_missing_translations:
-                # This translation seems to be new and should be added
-
-                add_missing_translation(
-                    key=self.key,
-                    hint=self.hint,
-                    default_text=self.defaultText,
-                    filename=self.filename,
-                    lineno=self.lineno,
-                    public=self.public,
-                )
+            if self.key not in systemTranslations:
+                # either the translate()-object has add_missing set
+                if not (add_missing := self.add_missing):
```

Comment:
I find this behaviour confusing.

- So I set `add_missing` set to `False`, but if `conf.i18n.add_missing_translations` is `True`, it is added anyway.
If `add_missing` is set to `True`, it doesn't matter what is in `conf.i18n.add_missing_translations` and it's added too.


I would see this as a common condition:
If `self.add_missing` and `conf.i18n.add_missing_translations` are both `True`, add, otherwise not.

Reply:
we can make it so.

Reply:
Hi @sveneberth, sorry for the confusion. The initial consideration about this flag was, that either the translate-object has the `add_missing`-flag on first order, and when this is not True, the globally defined logic of `conf.i18n.add_missing_translations` takes place, which can either be just True, but also one or many fnmatch-style pattern to match the given translation key.

Reply:
it seems we have different use cases for this and other exceptions (you want an `or` logic, I an `and`). Let's talk about this in a meet.

Reply:
As discussed yesterday, I have implemented Always and Never in 58d2e3057.